lib/m3u8-parser.js

```diff
@@ -22,6 +27,26 @@ var m3u8parser = module.exports = function() {
       self.emit('item', line.trim());
     }
   }
+  function ytdlParse() {
+    var time = self.ytdlSavedLines[self.ytdlSavedLines.length - 2];
+    if (!time) {
+      self.emit('ytdl-error', new Error('Time is null/undefined.\n' + time));
+      return;
+    }
+    time = parseInt(time.replace(/#EXTINF:/g, "").replace(/,/g, ""));
+    if (self.isFirstTime) time = 1;
+    var line = self.ytdlSavedLines[self.ytdlSavedLines.length-1];
+    if (!line) {
+      self.emit('ytdl-error', new Error('Line is null/undefined.\n' + time));
+      return;
+    }
+    parseLine(line);
+    var timeout = setTimeout(function() {
+      self.ytdlSavedLines = [];
+      return self.emit('ytdl-end');
+      clearTimeout(timeout); // Just in case timeout is still set, we want to make sure it's destroyed.
```

Comment:
I do realize this line is now redundant, I added it when there wasn't a return for the self.emit. I was having issues in this area, though, with it requesting 4-5 times a second. I don't know if this still happens, but I think it is happening very slowly. The longer you listen, the more you hear stuttering. Try adding a console.log here to see how many times it calls. It should only all every 5 seconds (or however long <time> is).

TL;DR: I believe I ended up creating a memory leak right here.